### PR TITLE
[Docs] A draft version of the contributing guideline document

### PR DESCRIPTION
**Pull request by kr11**
## Pull Request Description
Updated the `CONTRIBUTING.md`, which includes detailed procedures on how to deploy and how to submit code, among other updates.

## Related Issues
Resolves: #[46](https://github.com/aibrix/aibrix/issues/46)

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[x] PR title includes appropriate prefix(es)</li>
    <li>[x] Changes are clearly explained in the PR description</li>
    <li>[x] New and existing tests pass successfully</li>
    <li>[x] Code adheres to project style and best practices</li>
    <li>[x] Documentation updated to reflect changes (if applicable)</li>
    <li>[x] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,3 +1,195 @@
-# Contribution Guidelines
+## About AIBricks
 
-TODO
```
Comment (Jeffwan) [inline CONTRIBUTING.md:3]: We do not need to give the intro here. `README.md` can cover this part.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,3 +1,195 @@
...
+- **Kubernetes Documentation**: [Explore Kubernetes Docs](https://kubernetes.io/docs/home/) - Detailed explanations of Kubernetes concepts.
+
+## Prerequisites
+
+Before you start, ensure the following are installed on your system:
+
+- Go version 1.21.0 or higher
+- Docker version 17.03 or higher
+- kubectl version 1.11.3 or higher
+- Access to a Kubernetes cluster, version 1.11.3 or higher
```
Comment (Jeffwan) [inline CONTRIBUTING.md]: Let's change to 1.25.x or higher. 1.11 is too old. similar to kubectl

Comment (Jeffwan) [reply CONTRIBUTING.md]: Docker 17.03 seems to be old as well.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,3 +1,195 @@
...
+
+NOTE: The makefile target mentioned above generates an 'install.yaml' file in the dist directory. This file contains all the resources built with Kustomize, which are necessary to install this project without its dependencies.
+
+1. Using the installer: Users can just run `kubectl apply -f <URL for YAML BUNDLE>` to install the project, i.e.:
+
+```
+kubectl apply -f https://raw.githubusercontent.com/<org>/aibrix/<tag or branch>/dist/install.yaml
+```
+
+## How to Contribute
```
Comment (Jeffwan) [inline CONTRIBUTING.md:6]: I suggest to keep the content here in contributing guidance doc. rest of them are more like development.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,3 +1,195 @@
-# Contribution Guidelines
+## About AIBricks
 
-TODO
\ No newline at end of file
+AIBricks is a cloud-native solution designed to deploy, manage, and scale large language models (LLMs) efficiently, catering specifically to enterprise needs. For more information about AIBricks and its features, see [AIBricks Doc](https://github.com/aibrix/aibrix/tree/main/docs/tutorial) .
+
+## Important Links and Resources
```
Comment (Jeffwan) [inline CONTRIBUTING.md]: can we move line 5- line 149 to `DEVELOPMENT.md`

Comment (Jeffwan) [reply CONTRIBUTING.md]: Let's move `DEVELOPMENT.md` to `docs/development` as `README.md`

File: CONTRIBUTING.md (hunk 1 above)
Comment (kr11) [reply CONTRIBUTING.md:3]: Removed.

File: CONTRIBUTING.md (hunk 4 above)
Comment (kr11) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (kr11) [reply CONTRIBUTING.md]: I've updated the versions in `docs/development/README.md`. I upgraded Docker version to  `26.x` according to my env, is it suitable?

Comment (Jeffwan) [reply CONTRIBUTING.md]: sure. that works

**Pull request merged**